spec.emu

```diff
@@ -495,18 +671,27 @@ contributors: Ujjwal Sharma, Younies Mahmoud
         1. Let _r_ be ResolveLocale(%DurationFormat%.[[AvailableLocales]], _requestedLocales_, _opt_, %DurationFormat%.[[RelevantExtensionKeys]], %DurationFormat%.[[LocaleData]]).
         1. Let _locale_ be r.[[locale]].
         1. Set _durationFormat_.[[Locale]] to _locale_.
+        1. Set _durationFormat_.[[DataLocale]] to _r_.[[dataLocale]].
+        1. Let _dataLocale_ be _durationFormat_.[[DataLocale]].
+        1. Let _dataLocaleData_ be _durationFormat_.[[LocaleData]].[[&lt;_dataLocale_&gt;]].
+        1. Let _digitalFormat_ be _dataLocaleData_.[[DigitalFormat]].
```

Comment:
Basically this is the value you actually need, the few lines right before this one can be collapsed into this.

Reply:
I've thought pretty hard about this one. Collapsing these lines together doesn't match rest-of-402 style, as I understand it. See:

* steps 18 and 19 of [`Intl.DisplayNames` constructor](https://tc39.es/ecma402/#sec-Intl.DisplayNames):]

        18. Let _resolvedLocaleData_ be _r_.[[LocaleData]].
        19. Let _types_ be _resolvedLocaleData_.[[types]].

with _resolvedLocaleData_ never used after the [[types]] access.

* Steps 14 and 15 of [`Intl.ListFormat` constructor](https://tc39.es/ecma402/#sec-Intl.ListFormat), _resolvedLocaleData_ likewise not used after _resolvedLocaleData_.[[<type>]] access:

        14. Let _resolvedLocaleData_ be _r_.[[LocaleData]].
        15. Let _dataLocaleTypes_ be _resolvedLocaleData_.[[&lt;_type_&gt;]].


* First two steps of [`GetNumberFormatPattern` AO](https://tc39.es/ecma402/#sec-getnumberformatpattern) in `Intl.NumberFormat`

        1. Let _resolvedLocaleData_ be _numberFormat_.[[LocaleData]].
        2. Let _patterns_ be _resolvedLocaleData_.[[patterns]].
        
 (among others)

Additionally, the pattern "Let _alias_ be _obj_.[[X]].[[Y]].[[Z]]" is very rare in 402 algorithms; it shows up in one place in `Intl.DateTimeFormat` and nowhere else (https://tc39.es/ecma402/#sec-date-time-style-format)

Four-deep accesses (as would happen here if these were collapsed together) occur not at all in rest-of-402